test/drbgtest.c

```diff
@@ -824,66 +727,25 @@ static int test_rand_prediction_resistance(void)
         || !TEST_true(EVP_RAND_instantiate(z, 0, 0, NULL, 0, NULL)))
         goto err;
 
-    /*
-     * During a normal reseed, only the last DRBG in the chain should
-     * be reseeded.
-     */
-    inc_reseed_counter(y);
-    xreseed = reseed_counter(x);
-    yreseed = reseed_counter(y);
-    zreseed = reseed_counter(z);
-    if (!TEST_true(EVP_RAND_reseed(z, 0, NULL, 0, NULL, 0))
-        || !TEST_int_eq(reseed_counter(x), xreseed)
-        || !TEST_int_eq(reseed_counter(y), yreseed)
-        || !TEST_int_gt(reseed_counter(z), zreseed))
+    /* Fill the first buffer with random bytes */
+    if (!TEST_true(EVP_RAND_generate(z, buf1, sizeof(buf1), 0, 0, NULL, 0)))
         goto err;
 
     /*
      * When prediction resistance is requested, the request should be
      * propagated to the primary, so that the entire DRBG chain reseeds.
      */
```

Comment:
I'm not sure whether the tests still fulfill what the comment promises:

IMHO the upward propagation of prediction resistance requests is an important test, i.e. to check whether the reseeding happens all the way up and fresh entropy is requested. Does this test still exist after you changes? If not, it needs to be reimplemented.  Since the reseed counters are gone, the best way is probably to attach a test entropy source to the primary DRBG which is able count the requests.

Reply:
I was considering just dropping this test.  It's not testing anything anymore.